spec/models/abuse_report_spec.rb

```diff
@@ -38,6 +38,24 @@
       end
     end
 
+    context "missing work id" do
+      expected_url = ""
+      let(:missing_work_id) { 
```

Comment:
Don't know how get around this here

Reply:
You can use `do...end` instead of `{...}`. For example: https://github.com/otwcode/otwarchive/blob/d13d1c217534ee5714e4d413534b61298d561039/spec/helpers/bookmarks_helper_spec.rb#L275-L277

However, I don't think you need a multi-line block here. Something more like this would work:

```
    context "with a chapter URL that's missing the work id" do
      context "when the chapter exists" do
        # Work and chapter can be defined here and referenced when building our report.
        let(:work) { create(:work) }
        let(:chapter) { work.chapters.first }
        # We don't reference the URLs multiple times, so we can just enter type them out 
        # where needed.
        let(:missing_work_id) { build(:abuse_report, url: "http://archiveofourown.org/chapters/#{chapter.id}") }

        # While you can test the before_validation callback itself by calling it directly, or by
        # using validate, we've used save throughout this file to test validations, so let's stick with that. 
        it "saves and adds the correct work id to the URL" do
          expect(missing_work_id.save).to be_truthy
          expect(missing_work_id.url).to eq("http://archiveofourown.org/works/#{work.id}/chapters/#{chapter.id}/")
        end
      end

      context "when the chapter does not exist" do
        # We're using the same URL twice, so we're using let to memorize the value.
        let(:chapter_url) { "http://archiveofourown.org/chapters/000" }
        let(:missing_work_id) { build(:abuse_report, url: chapter_url) }

        it "saves without modifying the URL" do
          expect(missing_work_id.save).to be_truthy
          expect(missing_work_id.url).to eq(chapter_url)
        end
      end
    end
```

Reply:
Cheers man, I just used yours. Now I finally realize what save does. When I tried to do it the same way, I did it without save and just got the same URL I submitted haha. It's so hard without inline documentation damn